fails:IO.try_convert propagates an exception raised by #to_io
